surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/TestStrategyApplier.java

```diff
@@ -37,15 +42,45 @@ TestsToRun apply(Configuration configuration) {
     }
 
     private Set<Class<?>> selectTests(Configuration configuration) {
+        boolean reportDisabled = System.getProperty("smart.testing.report.disable") != null;
+        final List<Strategy> strategyList = new ArrayList<>();
         final List<String> strategies = Arrays.asList(configuration.getStrategies());
         final Set<TestSelection> selectedTests = new LinkedHashSet<>();
         for (final String strategy : strategies) {
             final TestExecutionPlanner plannerForStrategy = testExecutionPlannerLoader.getPlannerForStrategy(strategy);
-            selectedTests.addAll(plannerForStrategy.getTests());
+            final Collection<TestSelection> selections = plannerForStrategy.getTests();
+
+            selectedTests.addAll(selections);
+            if (!reportDisabled) {
+                strategyList.add(Strategy.newBuilder()
+                    .withStrategy(strategy)
+                    .withTests(selections.stream().map(TestSelection::getClassName).collect(Collectors.toList()))
+                    .build()
+                );
+            }
+
         }
         logger.info("Applied strategies: %s", strategies);
```

Comment:
I think it's not possible. because we are iterating over strategies to get all classes for that strategy in first if where we are building `Strategy` List.

Once we iterate over entire fore loop, we do have list of strategy ["new", "changed"] & list of Strategy built using builder in above for loop. so we can just simply build object with these values after for loop.